### PR TITLE
AutoUpdate Remote Certificate Validation Callback Backend

### PR DESCRIPTION
**Pull request by matthew17754**
This is the utility level implementation for the auto updater. I've held off on the front end UI implementation until the GUI scheme has been solidified. @KyroVibe Your call if we should come back to this later or merge and implement the front end later, as it shouldn't effect anything else in the code base. The areas for where the UI calls will be are commented out for the time being. 

This also renames `UnityHandles.cs`

### CODE REVIEW

File: engine/CoreEngine/Assets/Scripts/Engine/Util/UnityHandler.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        public static event CallBack OnFixedUpdate = () => { };
+        public static event CallBack OnLateUpdate = () => { };
+
+        void Start()
+        {
+            _ = SimulatorHandler.Instance;
+            DontDestroyOnLoad(gameObject);
+
+            string CurrentVersion = "4.3.3";
+            //GameObject.Find("VersionNumber").GetComponent<Text>().text = "Version " + CurrentVersion;
```
Comment (matthew17754) [inline engine/CoreEngine/Assets/Scripts/Engine/Util/UnityHandler.cs:32]: Finds the version number component and concatenates the version number to it.

File: engine/CoreEngine/Assets/Scripts/Engine/Util/UnityHandler.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                updater = update.URL;
+
+                var localVersion = new Version(CurrentVersion);
+                var globalVersion = new Version(update.Version);
+
+                var check = localVersion.CompareTo(globalVersion);
+
+                if (check < 0)
+                {
+                    //Auxiliary.FindGameObject("UpdatePrompt").SetActive(true);
```
Comment (matthew17754) [inline engine/CoreEngine/Assets/Scripts/Engine/Util/UnityHandler.cs:49]: Finds the update prompt game object and activates it.

File: engine/CoreEngine/Assets/Scripts/Engine/Util/UnityHandler.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        }
+
+        public void GetUpdate(bool yes)
+        {
+            if (yes)
+            {
+                Process.Start("http://bxd.autodesk.com");
+                Process.Start(updater);
+            }
+            //else Auxiliary.FindObject("UpdatePrompt").SetActive(false);
```
Comment (matthew17754) [inline engine/CoreEngine/Assets/Scripts/Engine/Util/UnityHandler.cs:128]: Finds the update prompt game object and deactivates it.

File: engine/CoreEngine/Assets/Scripts/Engine/Util/VersionManager.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assets.Scripts.Engine.Util
```
Comment (KyroVibe) [inline engine/CoreEngine/Assets/Scripts/Engine/Util/VersionManager.cs:7]: Change Assets.Scripts.Engine.Util to Synthesis.Util

**Pull request merged**